### DISCUSSION TITLE
Can you help optimize it

### DISCUSSION THREAD

**Discussion by vigee88** [ljleb/comfy-mecha#12] [Q&A] [answered]
I made a workflows myself to test the effect of each layer change in the model merge. However, halfway through the run, there was an error exit, indicating that the model cache had run out of memory. Can it be set to automatically delete the cache and release memory?
[SDXL分层的各层测试（完成）new new.json](https://github.com/user-attachments/files/16785640/SDXL.new.new.json)
     Thank you for your reply!

**Comment by ljleb** [comment, marked as answer]
Hi, the problem stems from the fact that comfyui caches everything by default. For model merging, this is very impractical.

There is a builtin boolean input "temporary_merge" on the Mecha Merger node that, when toggled on, automatically clears its own output as soon as its output is not used anymore for the rest of the workflow:

![image](https://github.com/user-attachments/assets/0d3fc648-ea5d-49ef-9e05-f3c59ca6fefd)

The node pack assumes that as soon as another merger starts executing, or at the end of the workflow, previous temporary merges can be deleted and this is where the cache clear occurs.

However, in this case, even though you left the checkbox toggled on, the node pack is unable to free the memory because the generated model is transformed by another node (the Load Lora node in this case).

So in other words, while the output of the merger is cleared by the node pack to save memory, the output of the lora node is not.

Visually, the node pack deletes these outputs:

![image](https://github.com/user-attachments/assets/b87a8a1d-e763-4b46-87f8-dfcdef7a1193)

But it does *not* delete these outputs:

![image](https://github.com/user-attachments/assets/adf5f616-6425-40ff-be08-c4d4868a4a84)

I suggest you use the builtin lora mecha recipe node along with the add difference recipe to merge the lora to the model during model loading, so that the cache clearing mechanism can effectively do its work. Something like this for your setup:

![image](https://github.com/user-attachments/assets/d3dc6543-5345-4c6d-83fa-3c903bb514ab)

**Comment by ljleb** [comment]
I wonder if it is possible and practical to have the merger node additionally clear outputs derived from the model it returns. I will make a note to come back to this later.

**Comment by vigee88** [comment]
Thank you so much. I have modified the workflow according to your instructions and it is now running smoothly. What a great program, with powerful functions.
![01](https://github.com/user-attachments/assets/fed444dc-18c7-4eb4-8570-fd2032ec7741)


[SDXL分层的各层测试（完成）new flnal-2.json](https://github.com/user-attachments/files/16792485/SDXL.new.flnal-2.json)
